Cython/Compiler/ExprNodes.py

```diff
@@ -501,26 +501,26 @@ def result(self):
             return self.calculate_result_code()
 
     def _make_move_result_rhs(self, result, allow_move=True, optional=True):
-        if not optional:
-            self.has_temp_moved = True
-            return "std::move({0})".format(result)
         if (self.is_temp and allow_move and
                 self.type.is_cpp_class and not self.type.is_reference):
             self.has_temp_moved = True
-            return "__PYX_STD_MOVE_IF_SUPPORTED({0})".format(result)
+            return "{}({})".format(
```

Comment:
The problem here is that because of the check for `self.type.is_cpp_class` above, this won't call `std::move()` on fundamental types, which we definitely need to allow.

We should safely be able to remove the check for `self.type.is_cpp_class`. I brought this up [here](https://github.com/cython/cython/pull/3821).

Reply:
Some of the cpp tests now fail :( Any objects to either:

1. Removing the `is_cpp_class` restriction, or
2. Reverting the latest commit

?

Reply:
If you do remove it then check that we don't generate `__PYX_MOVE_IF_SUPPORTED` in C mode. It'd probably be harmless (since it wouldn't be supported) but I don't think it should be there.

Reply:
I'm ok with either adding `… or is_rvalue_reference` or, preferably, always trying to `std::move()` whatever we have in C++ mode. The latter seems a more "intrusive" but probably also more correct solution. Especially in C++, user declared types might not always be exactly what Cython sees of them. Doing something that "usually doesn't hurt" but "might work better in some cases" seems a good path to walk.

Reply:
> always trying to std::move() whatever we have in C++ mode.

I'd agree this would be a good solution.

I [struggled for a while](https://github.com/cython/cython/pull/3821#issuecomment-693626570) to get a handle to the scope from inside this function or its callers in a straightforward way. I'm also not sure I understand what https://github.com/cython/cython/pull/3821#issuecomment-693635460 would look like. Any suggestions?